woocommerce/woocommerce-ios
# Jetpack CP: Check for Jetpack plugin status before installing and add retry functionality

Part of #5365 


### Description

This PR updates `JetpackInstallStepsViewModel` to improve the Jetpack install steps:
- Adds a check for Jetpack plugin details. If the check returns error, proceed to install plugin; otherwise activate the plugin if it's not already active.
- Updates `startInstallation` by checking Jetpack plugin details first.
- Updates error handling for installation and activation steps by checking for plugin details and retrying if necessary.
- Adds new `@Published` variable for failure of installation.
- Also: adds new activity indicator for when checking plugin details and installation step hasn't started.

### Testing instructions

Prerequisite: the account is connected to at least a JCP site (setup tips in p1636091588189400-slack-C6H8C3G23).
- Launch the app and select the Jetpack benefit banner.
- Proceed to install Jetpack. Notice that installation and activation steps still succeed. The process will stuck at "Connecting your store" due to issue mentioned in p91TBi-6EK-p2.
- Dismiss the screen and navigate back to Dashboard screen. Repeat steps 1 and 2. Notice that installation and activation steps are done immediately (since the plugin has been installed and activated earlier). 

### Screenshots

<img width="339" alt="Screen Shot 2021-12-01 at 13 41 49" src="https://user-images.githubusercontent.com/5533851/144185038-2286a38e-c232-441a-a0c8-79d648cd8a8a.png">

---
